src/lean_spec/subspecs/containers/signature.py

```diff
@@ -4,18 +4,22 @@
 
 from lean_spec.types import Bytes3100
 
+from ..xmss.containers import PublicKey
+from ..xmss.containers import Signature as XmssSignature
+from ..xmss.interface import DEFAULT_SIGNATURE_SCHEME
+
 
 class Signature(Bytes3100):
     """Represents aggregated signature produced by the leanVM (SNARKs in the future)."""
 
-    @staticmethod
-    def is_valid(signature: Signature) -> bool:
-        """Return True when the placeholder signature is the zero value."""
-        # TODO: Replace placeholder check once aggregated signatures are
-        # wired in as part of the multi-proof integration work.
-        return signature == Signature.zero()
-
-    @classmethod
-    def zero(cls) -> Signature:
-        """Return the zero (placeholder) signature."""
-        return cls(Bytes3100.zero())
+    def verify(self, public_key: PublicKey, epoch: int, message: bytes) -> bool:
+        """Verify the signature using XMSS verification algorithm."""
+        try:
+            scheme = DEFAULT_SIGNATURE_SCHEME
+            # Signature container is always 3100 bytes, but scheme config may expect less.
+            # Slice to the expected size if needed, assumes padding to the right.
+            signature_data = bytes(self)[: scheme.config.SIGNATURE_LEN_BYTES]
+            signature = XmssSignature.from_bytes(signature_data, scheme.config)
+            return scheme.verify(public_key, epoch, message, signature)
+        except Exception:
+            return False
```

Comment:
This is more severe than https://github.com/leanEthereum/leanSpec/pull/117/files#r2491285231 because it's part of the specs. But let's address separately.